### DISCUSSION TITLE
Next.js 13 Link and Chakra components

### DISCUSSION THREAD

**Discussion by abarani** [chakra-ui/chakra-ui#7479] [Q&A]
I came across the new version of the [Getting Started - Next.js Guide](https://chakra-ui.com/getting-started/nextjs-guide) documentation and noticed it recommends using the new `Link` component included in the `@chakra-ui/next-js` package to integrate Next.js 13 Link component.

I'm used to integrate Next.js 13 links using the Chakra `as` prop:
```typescript
import NextLink from "next/link";
import { Link as ChakraLink } from "@chakra-ui/react";

const Example = () => {
  return (
    <ChakraLink as={NextLink} href="/example">
      Example
    </ChakraLink>
  );
};
```

I'm a bit puzzled as the docs don't even mention the `as` prop as a solution to this.
It is even more flexible than the new Chakra integration `Link` component since it can also be used with other components, too (e.g. `Button` rendered as `a` tag).

Is there a reason why the new custom integration `Link` component is preferred to using the standard `as` prop?
Are there some edge cases covered by the component that I'm missing? 🤔

**Comment by synasapmob** [comment]
well, your idea is good but I guess props `as` just only extends what property of 'object',
example **ChakraLink** contains property 'sx/width/height' so  **NextLink* can use directly props of ChakraLink,
and if using `<ChakraLink bg='red' />` is available and else you cannot use too `<NextLink bg='red' />`

(sorry for my bad English)
